app/code/Magento/Sales/Test/Mftf/Test/AdminReorderAfterDeleteCustomerGroupTest.xml

```diff
@@ -0,0 +1,74 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+ /**
+  * Copyright © Magento, Inc. All rights reserved.
+  * See COPYING.txt for license details.
+  */
+-->
+<tests xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:noNamespaceSchemaLocation="urn:magento:mftf:Test/etc/testSchema.xsd">
+    <test name="AdminReorderAfterDeleteCustomerGroupTest">
+        <annotations>
+            <features value="Sales"/>
+            <stories value="Github issue: #29059 Error at reorder for order with no longer existing customer group id."/>
+            <title value="Unable to create 'Reorder'."/>
+            <description value="Unable to create 'Reorder' after deleting customer group."/>
+            <severity value="CRITICAL"/>
+            <group value="sales"/>
+        </annotations>
+
+        <before>
+            <createData entity="SimpleProduct2" stepKey="createSimpleProductApi"/>
+            <createData entity="Simple_US_Customer" stepKey="createCustomer" />
+            <actionGroup ref="AdminLoginActionGroup" stepKey="loginAsAdmin"/>
+        </before>
+
+        <after>
+            <deleteData createDataKey="createSimpleProductApi" stepKey="deleteSimpleProductApi"/>
+            <deleteData createDataKey="createCustomer" stepKey="deleteCustomer"/>
+            <actionGroup ref="AdminLogoutActionGroup" stepKey="logoutFromAdmin"/>
+        </after>
+
+        <actionGroup ref="AdminCreateCustomerGroupActionGroup" stepKey="createCustomerGroup">
+            <argument name="groupName" value="{{CustomCustomerGroup.code}}"/>
+            <argument name="taxClass" value="{{CustomCustomerGroup.tax_class_name}}"/>
+        </actionGroup>
+
+        <magentoCLI command="indexer:reindex" stepKey="reindex"/>
```

Comment:
To improve tests isolation and speeding up the performance of the functional test in general, there was an agreement to use the recently introduced CRON group for invoking the reindexing process. So I would strongly recommend it. Please, refer to the following  example 

https://github.com/magento/magento2/blob/b08c61d28875e03fbdd6f614ec6518c3afc48b38/app/code/Magento/Sales/Test/Mftf/Test/AddConfigurableProductToOrderFromShoppingCartTest.xml#L77